tools/cold-store/src/cli.rs

```diff
@@ -26,15 +29,30 @@ enum SubCommand {
     /// Copy n blocks to cold storage and update cold HEAD. One by one.
     /// Updating of HEAD happens in every iteration.
     CopyNextBlocks(CopyNextBlocksCmd),
+    /// Copy all blocks to cold storage and update cold HEAD.
+    CopyAllBlocks(CopyAllBlocksCmd),
 }
 
 impl ColdStoreCommand {
     pub fn run(self, home_dir: &Path) {
-        let near_config = nearcore::config::load_config(
-            &home_dir,
-            near_chain_configs::GenesisValidationMode::Full,
-        )
-        .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+        let near_config = {
+            let mut near_config = nearcore::config::load_config(
+                &home_dir,
+                near_chain_configs::GenesisValidationMode::Full,
+            )
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+            // Before copying all blocks create a checkpoint and open it as hot db.
+            if let SubCommand::CopyAllBlocks(_) = self.subcmd {
+                near_config.config.store = create_checkpoint_for_cold_copy(
+                    &near_config.config.store,
+                    home_dir,
+                    &"data/data-checkpoint",
+                )
+                .unwrap();
+            }
+            near_config
+        };
```

Comment:
Can you move this to a helper method and only call it in the CopyAllBlocks match arm?